File: MyProjects/MAP/Assignment 7/Interpreter/Model/Statements/IfStmt.cs
using System;
using System.Collections.Generic;
using System.Text;
using Interpreter.Model.Expressions;

namespace Interpreter.Model.Statements
{
    public class IfStmt:IStmt
    {
        private IExpr expr;
        private IStmt thenStmt;
        private IStmt elseStmt;

        public IfStmt(IExpr expression, IStmt thenS, IStmt elseS)
        {
            expr = expression;
            thenStmt = thenS;
            elseStmt = elseS;
        }

        public ProgramState execute(ProgramState state)
        {
            int val = expr.evaluate(state.SymbolTable);

            if (val != 0)
                state.Stack.add(thenStmt);
            else
                state.Stack.add(elseStmt);
            return state;
        }

        public override string ToString()
        {
            return "if(" + expr + ") then(" + thenStmt + ")else(" + elseStmt + ")";
        }
    }
}

File: MyProjects/MAP/Practical Exam/Procedures/src/Model/Statements/IfStmt.java
package Model.Statements;

import  Model.Utilities.*;
import Model.ProgramState;
import Model.Expressions.*;

public class IfStmt implements IStmt {
    private IExpr expr;
    private IStmt thenStmt;
    private IStmt elseStmt;

    public IfStmt(IExpr expression, IStmt thenS, IStmt elseS){
        expr = expression;
        thenStmt = thenS;
        elseStmt = elseS;
    }
    @Override
    public ProgramState execute(ProgramState state){
        IStack<IDictionary<String, Integer>> dic = state.getSyms();
        IStack<IStmt> stack = state.getStack();
        IDictionary<Integer, Integer> heap=state.getHeap();
        int val = expr.evaluate(dic,heap);

        if(val != 0)
            stack.add(thenStmt);
        else
            stack.add(elseStmt);
        return null;
    }

    public String toString(){
        return "if(" + expr + ") then(" + thenStmt +  ")else(" + elseStmt + ")";
    }
}
